src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -162,25 +168,40 @@ private static ProcessModuleCollection GetModules(int processId, bool firstModul
                             BaseAddress = ntModuleInfo.BaseOfDll
                         };
 
-                        int length = Interop.Kernel32.GetModuleBaseName(processHandle, moduleHandle, chars, chars.Length);
+                        int length = 0;
+                        while ((length = Interop.Kernel32.GetModuleBaseName(processHandle, moduleHandle, chars, chars.Length)) == chars.Length)
+                        {
+                            ArrayPool<char>.Shared.Return(chars);
+                            chars = ArrayPool<char>.Shared.Rent(length * 2);
```

Comment:
```suggestion
                            char[] toReturn = chars;
                            chars = ArrayPool<char>.Shared.Rent(length * 2);
                            ArrayPool<char>.Shared.Return(toReturn);
```

Reply:
what is the benefit of having a temporary local variable here?

Reply:
If Return were to throw an exception after successfully storing the array, your finally block will double-return it again.

Reply:
@stephentoub thanks!